datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -563,13 +566,22 @@ impl DisplayAs for ParquetSink {
 impl ParquetSink {
     /// Create from config.
     pub fn new(config: FileSinkConfig) -> Self {
-        Self { config }
+        Self {
+            config,
+            written: Default::default(),
+        }
     }
 
     /// Retrieve the inner [`FileSinkConfig`].
     pub fn config(&self) -> &FileSinkConfig {
         &self.config
     }
+
+    /// Retrieve the file metadata from the last write.
+    pub fn written(&self) -> Vec<FileMetaData> {
+        self.written.lock().clone()
+    }
```

Comment:
I think the comments here are out of date (it returns the FileMetaData written for *all* written files, not just the last one)

As mentioned above, without any additional information I don't think there is any way to know how to match up the written files and the entries in the `Vec<FileMetaData>` without some more information

However, if we changed the API to return a HashMap I think it would work well